Sources/Classes/RSServerConfigManager.m

```diff
@@ -114,6 +114,15 @@ - (RSServerConfigSource *)_parseConfig:(NSString *)configStr {
             destination.destinationConfig = [destinationDict objectForKey:@"config"];
             [destinations addObject:destination];
         }
+        
+        NSDictionary *dataResidencyUrls = [sourceDict objectForKey:@"dataPlaneUrls"];
+        for (NSString *key in [dataResidencyUrls allKeys]) {
+            if([[key lowercaseString] isEqualToString:@"us"]) {
```

Comment:
This parsing doesn't make sense here. Again you are finding the URL in RSServerConfigSource. Simply assign the `dataResidencyUrls` to `source.dataResidencyUrls`. Parse the urls in RSServerConfigSource and use `switch`.

Reply:
The reason why I am parsing here is that irrespective of whether we received the region keys from `sourceConfig` in `Capitals`  (or) `LowerCase` we are retrieving them and then in the case of RSServerConfigSource we are simply Storing the `Enum` to `URL` mapping. 

We are not iterating through the entire dictionary again in the RSServerConfigSource, we are just checking if a `RSDataResidencyServer`  enum exists, if yes returning the data plane URL.

We are cleaning/processing the data from the sourceConfig in the `_parseConfig` of `RSServerConfigManager` and then making it ready to use it for the `RSServerConfigSource`.

Please let me know, if this makes sense now

Reply:
> The reason why I am parsing here is that irrespective of whether we received the region keys from sourceConfig in Capitals (or) LowerCase we are retrieving them and then in the case of RSServerConfigSource we are simply Storing the Enum to URL mapping.
Here, you are parsing & storing again; it doesn't make sense.

Reply:
As discussed,  handling only the lowercase region values from the sourceConfig for now, and hence remove the parsing logic completely.